projects/plugins/social/src/class-meta-tags.php

```diff
@@ -237,6 +237,26 @@ public function get_description( $description = '', $data = null ) {
 		return $description;
 	}
 
+	/**
+	 * To set a custom OG:title for social notes.
+	 */
+	public function get_og_title_for_social_notes() {
+		$text   = wp_strip_all_tags( get_the_excerpt() );
+		$length = 55;
+		if ( strlen( $text ) <= $length ) {
+			return $text;
+		}
+
+		$trimmed_text   = substr( $text, 0, $length );
+		$last_space_pos = strrpos( $trimmed_text, ' ' );
+
+		if ( $last_space_pos !== false ) {
+			$trimmed_text = substr( $trimmed_text, 0, $last_space_pos );
+		}
```

Comment:
I wonder how this would work with other word break characters. Do you think we could use [str_word_count](https://www.php.net/manual/en/function.str-word-count.php), or maybe mb_str_word_count if it's available, with 2 as the second argument so we get the index of the word? We could cycle through the array and look for an `index + word length` that's greater than the desired length.

Reply:
I tried this out and everything was working perfectly until I tested it with a chatgpt generated essay `Empathy, often regarded as the cornerstone of human connection, holds immense potential in shaping societies into compassionate and resilient communities. Defined as the ability to understand and share the feelings of others, empathy transcends cultural, geographical, and ideological boundaries, fostering bonds of understanding and support among individuals. In today's fast-paced and interconnected world, cultivating empathy is not merely a virtue but a necessity for building cohesive and inclusive communities.` 

`str_word_count` doesn't retain the punctuation. The comma gets lost.